.github/scripts/bindist.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -eux
+
+. .github/scripts/env.sh
+. .github/scripts/common.sh
+
+# ensure ghcup
+if ! command -v ghcup ; then
+	install_ghcup
+fi
+
+# create tarball/zip
+case "${TARBALL_EXT}" in
+    zip)
+		HLS_VERSION="$(grep '^version:' haskell-language-server.cabal | awk '{ print $2 }')"
```

Comment:
I'd be tempted to make this a parameter and have a separate step for getting it. It's a little surprising that it gets pulled out in here, it's not where I'd look!